src/roiextractors/multiimagingextractor.py

```diff
@@ -56,18 +56,30 @@ def __init__(self, imaging_extractors: list):
         self._num_frames = int(num_frames)
 
     def get_frames(self, frame_idxs: ArrayType, channel: int = 0) -> NumpyArray:
-        extractor_index = self._get_imaging_extractor_index(frame_idxs[0])
-        rel_frame_index = frame_idxs[0] - int(self._start_frames[extractor_index])
+        extractor_indices = np.searchsorted(self._end_frames, frame_idxs, side="right")
+
+        frames_to_concatenate = []
+        # Extract frames for each extractor and concatenate
+        for extractor_index in extractor_indices:
+            frames_for_each_extractor = self._get_frames_from_an_imaging_extractor(
+                extractor_index=extractor_index,
+                frame_idxs=frame_idxs,
+            )
+            frames_to_concatenate.append(frames_for_each_extractor[np.newaxis, ...])
+
+        frames = np.concatenate(frames_to_concatenate, axis=0)
```

Comment:
this is for handling cases when the requested frames belong to different imaging extractors, but I doubt this is the final solution, I'll come back to it tomorrow!